queries.sql

```diff
@@ -7,4 +7,40 @@ SELECT date_of_birth FROM animals WHERE name='Agumon' OR name='Pikachu';
 SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name <> ' Gabumon';
-SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
\ No newline at end of file
+SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+-- update tables with transactions
+BEGIN TRANSACTION;
+UPDATE animals
+SET species = 'unspecified';
+ROLLBACK;
+
+-- DELETE COLUMN
+ALTER TABLE animals DELETE COLUMN species;
+
+-- update table in a transaction
+BEGIN;
+UPDATE animals SET species='digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+COMMIT;
+
+-- delete transaction in the table and roll back;
+DELETE FROM animals;
+ROLLBACK;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT one;
+UPDATE animals SET weight_kg = weight_kg * (-1);
+ROLLBACK TO SAVEPOINT one;
+UPDATE animals SET weight_kg = weight_kg * (-1) WHERE weight_kg < 0;
+COMMIT;
+
+-- WRITE COMMITPOINTQUERIES FOR QUESTIONS
+
+SELECT COUNT(name) AS total_animals FROM animals; 
+SELECT name, SUM(escape_attempts) FROM animals GROUP BY name having MIN(escape_attempts) < 1;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, MAX(escape_attempts) FROM animals GROUP BY neutered;
+SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals GROUP BY species;
+SELECT species, AVG(escape_attempts) FROM animals WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-01-01' GROUP BY species;
```

Comment:
- [ ] Kindly make sure the end date is 2000-12-31 as its the entire 2000 year.

Reply:
well resolved! ✔️